staging/src/k8s.io/apiserver/pkg/endpoints/handlers/delete.go

```diff
@@ -178,11 +178,6 @@ func DeleteCollection(r rest.CollectionDeleter, checkBody bool, scope *RequestSc
 			return
 		}
 
-		// enforce a timeout of at most requestTimeoutUpperBound (34s) or less if the user-provided
-		// timeout inside the parent context is lower than requestTimeoutUpperBound.
-		ctx, cancel := context.WithTimeout(ctx, requestTimeoutUpperBound)
-		defer cancel()
-
```

Comment:
Let's leave a comment why we actually don't set the timeout on the context.

Also - is it possible to add a simple unit test to prove that it won't get re-added later?

Reply:
I added a unit test, it makes sure that the `ctx` passed to  `DeleteCollection` does not have any timeout enforced.